html/mycakeapp/src/Model/Table/CreditcardsTable.php

```diff
@@ -107,4 +151,52 @@ public function buildRules(RulesChecker $rules)
 
         return $rules;
     }
+    protected function _initializeSchema(Schema $schema)
+    {
+        parent::_initializeSchema($schema);
+        $schema->columnType('card_number', 'EncryptedType');
+        return $schema;
+    }
+    public function findAllCardNumbers(Query $query, array $options)
+    {
+        $allCardNumbersArray = $query->select(['card_number', 'deadline'])->where([
+            'is_deleted' => 0,
+        ])->toArray();
+        //date_parse_from_formatを使うと月の頭の0が消えるため月にintをかけている
+        $today = date('Y') . (int)date('m');
+        foreach ((array)$allCardNumbersArray as $allCardNumberArray) {
+            $deadlineInfo = date_parse_from_format('m/y', $allCardNumberArray->deadline);
+            $deadline = $deadlineInfo['year'] . $deadlineInfo['month'];
+            if ((int)$deadline > (int)$today) {
```

Comment:
有効期限が今月のカードもまだ使えるので、今月分は含めた方がいいと思います！
```suggestion
            if ((int)$deadline >= (int)$today) {
```